### ISSUE TITLE
Unclosed file corrupted on quit

### ISSUE THREAD

**Issue by starsinmypockets** [Dimowner/AudioRecorder#21]
I have a large file that was not closed and is now corrupted. Is it possible to periodically save the file while it is recording in order to avoid losing large recordings entirely if the closed properly?

Also, I am looking for information how to recover the file. In particular, I am looking for information about how what encoding the audio is.

**Comment by Dimowner** [comment]
The app has 2 recording formats: 
By default is selected .m4a MPEG4 media file format encoded with AAC Low Complexity (AAC-LC) audio codec.
The second format WAV format it is pure PCM uncompressed audio data

The app has problems with recording long records. In the near future will fix these issues.

**Comment by PontiacCZ** [comment]
What recording length causes troubles in the current version? Half an hour? Two hours? Just curious so I don't end up with a corrupted file as the OP did. :-)

**Comment by Dimowner** [comment]
For now, almost all recording troubles are fixed.
 - Added available space tracking and if not available space on the device the recording will stop automatically.
 - Also, there was changed saving algorithm. Now when you start recording a new record will be created in the database and it will be updated when recording finish.
- Also fixed OutOfMemory crash during decoding waveform.

I've tested on a few devices recorded 8-9hour records successfully.
The only remaining issue is a waveform for long records (2-3hours and longer) are not displayed correctly usually it is a flat line.
